### DISCUSSION TITLE
Peppol Reporting Database configuration

### DISCUSSION THREAD

**Discussion by digitalantiz-team** [phax/phase4#195] [Q&A] [answered]
Dear @phax ,

Really appreciated your quick reply on my previous question regarding to Phase4 configuration, it's really helpful for me to come to a solution with your suggestion.

Now I'm encountering again another issue also related to configuration, it's Peppol reproting database configuration. Because I cannot use IConfig which passed into initBackend method as a parameter. One of suggestion that you gave me earlier is to use Environment Variable to override CONFIG_FILE, but this cannot work in my project due to some limitation in the deployment. I also tried to pass configuration to SPI implementation class by using @Autowired and @Value, but the values in application.properties cannot be read (I guess due to some loading order).

Any suggestion for me on this part will be great appriciated. 

Thank you so much again for your great work.

**Comment by phax** [comment, marked as answer]
@digitalantiz-team Thanks for reaching out to me. Most likely you used a different user name for the last query :)

`IConfig` is an interface that is implemented by e.g. class `com.helger.config.Config`. It's a abstration of configuration sources like system properties, environment variables, properties files or even `Map`s with keys and values.

See e.g. https://github.com/phax/ph-commons/blob/master/ph-config/src/test/java/com/helger/config/ConfigTest.java#L172  for a test function how to create a configuration based on a `Map`.

hth

**Reply by digitalantiz-team** [reply]
Hi @phax ,
Exactly what I'm looking for. Thank you so much for your quick reply, reallly appreciated again.

PS: yes, you're right, I have different github accounts, and I totally unawareness of using different username for the this question.

**Reply by phax** [reply]
Super. Feel free to click "Mark as answer" :)
